impl/src/testing.rs

```diff
@@ -102,13 +142,21 @@ pub fn make_basic_workspace() -> (TempDir, CargoWorkspaceFiles) {
   make_workspace(basic_toml(), Some(basic_lock()))
 }
 
+pub fn make_advanced_workspace() -> (TempDir, CargoWorkspaceFiles) {
+  make_workspace(advanced_toml(), Some(advanced_lock()))
+}
+
 /**
  * Configures the given mock_server (representing a crates.io remote) to return
  * mock responses for the given crate and version .
  */
-pub fn mock_remote_crate(name: &str, version: &str, mock_server: &MockServer) -> TempDir {
+pub fn mock_remote_crate<'server>(
+  name: &str,
+  version: &str,
+  mock_server: &'server MockServer,
+) -> (TempDir, Vec<MockRef<'server>>) {
```

Comment:
This is a pattern that existed in the project before I started working on this project: https://github.com/google/cargo-raze/blob/master/impl/src/testing.rs#L72-L99

Could this not be considered acceptable in test code? I'm happy to update documentation for this function to explain the return type.

Reply:
Hmm.

That is an entirely valid and powerful rebuttal, but I think that case is slightly different because it returns two self-descriptive types. These are not quite self descriptive.

Reply:
Updated